nipype/interfaces/ants/preprocess.py

```diff
@@ -0,0 +1,358 @@
+                format_args[metric_arg] = getattr(self.inputs, metric_arg)
+        return metric_str.format(**format_args)
+
+    def _format_transform(self):
+        transform_str = "-t {}[{}]"
+        if (_extant(self.inputs.transformation_model)
+                and _extant(self.inputs.gradient_step_length)):
+            return transform_str.format(self.inputs.transformation_model,
+                                        self.inputs.gradient_step_length)
+        raise ValueError("Unable to format transformation_model argument")
+
+    def _format_output(self):
+        if (_extant(self.inputs.output_transform_prefix)
+                and _extant(self.inputs.output_warped_image)
+                and _extant(self.inputs.output_average_image)):
+            return "-o [{},{},{}]".format(
+                self.inputs.output_transform_prefix,
+                self.inputs.output_warped_image,
+                self.inputs.output_average_image
+            )
+        elif _extant(self.inputs.output_average_image):
+            return "-o {}".format(self.inputs.output_average_image)
+        else:
+            raise ValueError("Unable to format output due to lack of inputs.")
+
+    def _list_outputs(self):
+        outputs = self._outputs().get()
+        if _extant(self.inputs.output_average_image):
+            outputs['average_image'] = (
+                os.path.abspath(self.inputs.output_average_image)
+            )
+        if _extant(self.inputs.output_warped_image):
+            outputs['warped_image'] = (
+                os.path.abspath(self.inputs.output_warped_image)
+            )
+        if _extant(self.inputs.output_transform_prefix):
+            fname = '{}MOCOparams.csv'.format(
+                self.inputs.output_transform_prefix
+            )
+            outputs['composite_transform'] = os.path.abspath(fname)
+        if (_extant(self.inputs.write_displacement) and
+                _extant(self.inputs.output_transform_prefix) and
+                self.inputs.write_displacement is True):
+            fname = '{}Warp.nii.gz'.format(self.inputs.output_transform_prefix)
+            outputs['displacement_field'] = os.path.abspath(fname)
+        return outputs
+
+class AntsMatrixConversionInputSpec(BaseInterfaceInputSpec):
+    matrix = File(
+        exists=True,
+        desc='Motion crrection matrices to be converted into FSL style motion parameters',
+        mandatory=True
+    )
+
+class AntsMatrixConversionOutputSpec(TraitedSpec):
+    parameters = File(exists=True, desc="parameters to be output")
+
+
+class AntsMatrixConversion(BaseInterface):
+    '''
+    Take antsMotionCorr motion output as input, convert to FSL style
+    parameter files. Currently does not output origin of rotation.
+    '''
+    input_spec = AntsMatrixConversionInputSpec
+    output_spec = AntsMatrixConversionOutputSpec
+
+    def _run_interface(self, runtime):
+        in_fp = open(self.inputs.matrix)
```

Comment:
I would use here `np.loadtxt`, defining the `sep=','`. This way you get an numpy array, removing all the float casts down below (just one `.astype(float)` would be necessary).

Additionally, you can use a `with` here right?, is the handle necessary at all during the method?